edx_django_utils/monitoring/code_owner/middleware.py

```diff
@@ -19,35 +19,96 @@ class CodeOwnerMetricMiddleware:
     Custom metrics set:
     - code_owner: The owning team mapped to the current view.
     - code_owner_mapping_error: If there are any errors when trying to perform the mapping.
-    - view_func_module: The __module__ of the view_func, which can be used to
-        find missing mappings.
+    - code_owner_path_error: The error mapping by path, if code_owner isn't found in other ways.
+    - code_owner_path_module: The __module__ of the view_func which was used to try to map to code_owner.
+        This can be used to find missing mappings.
+    - code_owner_transaction_error: The error mapping by transaction, if code_owner isn't found in other ways.
+    - code_owner_transaction_name: The current transaction name used to try to map to code_owner.
+        This can be used to find missing mappings.
 
     """
     def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
-        self._set_owner_metrics_for_request(request)
         response = self.get_response(request)
+        self._set_code_owner_metric(request)
         return response
 
-    def _set_owner_metrics_for_request(self, request):
+    def process_exception(self, request, exception):
+        pass
```

Comment:
**Reviewer Note:** Updating this to be `self._set_code_owner_metric(request)` and adding some additional testing.

Reply:
This has been fixed.